src/DynamicRupture/FrictionLaws/GpuImpl/LinearSlipWeakening.h

```diff
@@ -24,65 +24,203 @@ class LinearSlipWeakeningBase : public GpuBaseFrictionLaw {
                                       seissol::initializers::DynamicRupture* dynRup,
-                                      real fullUpdateTime) override;
+                                      real fullUpdateTime) override {
+    auto* concreteLts = dynamic_cast<seissol::initializers::LTS_LinearSlipWeakening*>(dynRup);
+    this->dC = layerData.var(concreteLts->dC);
+    this->muS = layerData.var(concreteLts->muS);
+    this->muD = layerData.var(concreteLts->muD);
+    this->cohesion = layerData.var(concreteLts->cohesion);
+  }
 
   /**
    *  compute the slip rate and the traction from the fault strength and fault stresses
    *  also updates the directional slip1 and slip2
    */
-  void calcSlipRateAndTraction(FaultStresses& faultStresses,
-                               TractionResults& tractionResults,
-                               std::array<real, misc::numPaddedPoints>& strength,
-                               unsigned int timeIndex,
-                               unsigned int ltsFace);
+  void calcSlipRateAndTraction(FaultStresses* faultStressesPtr,
+                               TractionResults* tractionResultsPtr,
+                               real (*strengthBuffer)[misc::numPaddedPoints],
+                               unsigned int timeIndex) {
+
+    auto layerSize{this->currLayerSize};
+    auto* initialStressInFaultCS{this->initialStressInFaultCS};
+    auto* impAndEta{this->impAndEta};
+    auto* slipRateMagnitude{this->slipRateMagnitude};
+    auto* slipRate1{this->slipRate1};
+    auto* slipRate2{this->slipRate2};
+    auto* traction1{this->traction1};
+    auto* traction2{this->traction2};
+    auto* slip1{this->slip1};
+    auto* slip2{this->slip2};
+    auto* deltaT{this->deltaT};
+
+    #pragma omp target teams loop                \
+    map(to: deltaT[0:CONVERGENCE_ORDER])         \
+    is_device_ptr(faultStressesPtr,              \
+                  tractionResultsPtr,            \
+                  strengthBuffer,                \
+                  initialStressInFaultCS,        \
+                  impAndEta,                     \
+                  slipRateMagnitude,             \
+                  slipRate1,                     \
+                  slipRate2,                     \
+                  traction1,                     \
+                  traction2,                     \
+                  slip1,                         \
+                  slip2)                         \
+    device(diviceId)
+    for (unsigned ltsFace = 0; ltsFace < layerSize; ++ltsFace) {
+      auto& faultStresses = faultStressesPtr[ltsFace];
+      auto& tractionResults = tractionResultsPtr[ltsFace];
+      auto& strength = strengthBuffer[ltsFace];
+
+      #pragma omp loop bind(parallel)
+      for (unsigned pointIndex = 0; pointIndex < misc::numPaddedPoints; pointIndex++) {
+        // calculate absolute value of stress in Y and Z direction
+        real totalStressXY = initialStressInFaultCS[ltsFace][pointIndex][3] +
```

Comment:
traction1, traction2

Reply:
No I cannot. We already use these variable names here: https://github.com/SeisSol/SeisSol/blob/a8b039ea9fd2f7a148de2ea5b6230f31a2b30317/src/DynamicRupture/FrictionLaws/GpuImpl/LinearSlipWeakening.h#L73-L74

I renamed then to `totalStress1` and `totalStress2`. Is it ok? What do you think?